mono/utils/flags.c

```diff
@@ -0,0 +1,221 @@
+/**
+ * \file Runtime flags
+ *
+ * Copyright 2020 Microsoft
+ * Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+
+#include <stdio.h>
+
+#include "flags.h"
+#include "mono/utils/mono-error-internals.h"
+
+typedef enum {
+	MONO_FLAG_BOOL,
+	MONO_FLAG_BOOL_READONLY,
+	MONO_FLAG_INT,
+	MONO_FLAG_STRING
+} MonoFlagType;
+
+/* Define flags */
+#define DEFINE_FLAG_FULL(flag_type, ctype, c_name, cmd_name, def_value, comment) \
+	ctype mono_##c_name = def_value;
+#define DEFINE_FLAG_READONLY(flag_type, ctype, c_name, cmd_name, def_value, comment)
```

Comment:
shouldn't it be this?
```suggestion
#define DEFINE_FLAG_READONLY(flag_type, ctype, c_name, cmd_name, def_value, comment) \
	const ctype mono_##c_name = def_value;
```

Reply:
hmm, I just saw it's done in `flags.h`.  Not sure why those flags a defined again here.

Reply:
No, its already defined in flags.h as a static const.

Reply:
ah, it's due to `MONO_API_DATA`. nevermind 🙂